docs/manually_regen_sdk.md

```diff
@@ -0,0 +1,40 @@
+  3. Change the emitter version used in our `azure-sdk-for-net` repo in this file `eng\emitter-package.json` like this:
+  ```diff
+  {
+    "main": "dist/src/index.js",
+    "dependencies": {
+  -   "@azure-tools/typespec-csharp": "0.1.11-beta.20230212.4"
+  +   "@azure-tools/typespec-csharp": "/absolute/path/to/src/CADL.Extension/Emitter.Csharp/azure-tools-typespec-csharp-x.y.z.tgz"
+    }
+  }
+  ```
+  4. Locate the `dll` file generated for the generator, usually it could be found `autorest.csharp` repository in this directory: `artifacts\bin\AutoRest.CSharp\Debug\net6.0\AutoRest.CSharp.dll`
+- At this point there are several ways to generate but the idea is to generate and test all projects that will be affected by your PR.  There are tools such as [this](https://github.com/ArcturusZhang/Regen) which are written to handle specific cases.  If you are unsure you should run against all projects to be safe.
+
+To regen and test everything in azure-sdk-for-net after you have updated to use your new local build do the following:
+
+- First generate all projects in the repo by executing `dotnet build [RepoRoot]/eng/service.proj /t:GenerateCode`. If your generator PR needs to apply to typespec/cadl projects, or your generator PR changes anything in our emitter (the `src\CADL.Extension\Emitter.Csharp` project), you will need to run `dotnet build [RepoRoot]/eng/service.proj /t:GenerateCode  /p:typespecAdditionalOptions="csharpGeneratorPath=/absolute/path/to/artifacts/bin/AutoRest.CSharp/Debug/net6.0/AutoRest.CSharp.dll`.
+- Next we want to at minimum run the tests against the new generated code by using `dotnet test [RepoRoot]/eng/service.proj --filter "(TestCategory!=Manually) & (TestCategory!=Live)"`
+- For non GA libraries there could be API changes so we want to run the Export-API script with no parameters which will update any projects that now have an API change `[RepoRoot]\eng\scripts\Export-API.ps1`
+- Finally it is very possible that we will need to make test case changes or snippet changes especially for non GA libraries which have expected changes.  All of these should be made in the branch and included in the PR to demonstrate all resulting changes from the autorest.csharp PR.
+
+Once all of that is successful, we can commit all of the local changes except the NuGet file and `Packages.Data.props` and push those to a branch.  Once this is done we want to ensure this regen PR is discoverable as well as the key stakeholders who would need to sign off mark this PR as either approved or request changes.
+
+- In the autorest.csharp PR description please add a line at the very top indicating which azure-sdk-for-net PR is the regeneration for this PR.  An example would look like this
+![PR example image](./docs/images/regenPRLink.png)
+- Next have any stakeholders that would need to review do so in this PR and mark it as approved or request changes.
```

Comment:
Move this two line into `readme.md`. They are not steps of `regen`, they are  common steps for PR validation after both `manual regen` and `auto regen`